[TASK] Migrate to Vite

weeman1337 opened the pull request:
- Migrates to
  - Vite
  - ES modules
- Adds prettier :lipstick: 
- Migrates to ESLint recommended

closes https://github.com/freifunk/meshviewer/issues/13

## Description + Motivation and Context

This PR migrates the build system from Gulp to Vite. Gulp is rarely maintained, the packages become outdated and it becomes more and more problematic to keep the dependencies up to date. Good example are [the overrides](https://github.com/freifunk/meshviewer/blob/main/package.json#L65) to „fix“ the dependencies security issues.

Because this PR touches every file anyway :shrug: It also introduces Prettier for a consistent code-style.

## How Has This Been Tested?

Tested builds locally.

## Screenshots/links:

No user-facing change

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project. (CI will test it anyway and also needs approval)
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.

MyIgel commented on README.md:
````diff
@@ -1,49 +1,61 @@
+
 ```bash
-docker  run -it --rm -u $(id -u):$(id -g) -v "$PWD":/app -w /app -e NODE_ENV=development -p 3000:3000 node npm run gulp serve
+docker run -it --rm -u $(id -u):$(id -g) -v "$PWD":/app -w /app -e NODE_ENV=development -p 3000:3000 node npm run dev
````
```suggestion
docker run -it --rm \
  -u $(id -u):$(id -g) -v "$PWD":/app -w /app \
  -e NODE_ENV=development -p 5173:5173 node \
  npm run dev -- \
  --host 0.0.0.0
```

MyIgel commented on README.md:
````diff
@@ -1,49 +1,61 @@
+docker run -it --rm -u $(id -u):$(id -g) -v "$PWD":/app -w /app -e NODE_ENV=development -p 3000:3000 node npm run dev
 ```
+
 The map is reachable at [localhost:3000](http://localhost:3000).
````
```suggestion
The map is reachable at [localhost:5173](http://localhost:5173).
```

MyIgel reviewed (changes requested):
That setup works so great! :heart_eyes: 

Some preliminary findings:

* npm install shows warnings, can they easily be fixed?
  ```
  npm WARN deprecated @stylelint/postcss-markdown@0.36.2: Use the original unforked package instead: postcss-markdown
  npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
  npm WARN deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
  npm WARN deprecated @stylelint/postcss-css-in-js@0.37.3: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
  npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
  ```

* There seem to be some possible moderate vulnerabilities, are they easily fixable? 

* When opening a node the left side panel is blank and the console shows an error.

* When opening the node graph, the page seems to work but shows an error in the console

either way it might have been easier to recview if the linting vs migration would have been two MRs ;)

MyIgel commented on lib/forcegraph.js:
```diff
@@ -1,288 +1,292 @@
+import * as d3Selection from "d3-selection";
+import * as d3Timer from "d3-timer";
+import * as d3Zoom from "d3-zoom";
+
```
```suggestion
import * as d3Ease from "d3-ease";
import * as d3Interpolate from "d3-interpolate";

```

MyIgel commented on lib/infobox/node.js:
```diff
@@ -1,239 +1,249 @@
+  return helper.showHwImg(V, o, subst);
+}
+
+export const Node = (el, d, linkScale, nodeDict) => {
```
This produces an error as it can't be constructed using `new Node(...)`

```suggestion
export function Node(el, d, linkScale, nodeDict) {
```

MyIgel reviewed (commented):
Those "fixes" might fix the above errors but might be implemented somehow better, dunno :D

Also I'm wondering if there is an easy option to keep the whole graph centred all the time, even before it gets rescaled to zoom to a specific node?

weeman1337 commented:
Errors are fixed. Removed and updated some packages.

Two deprecation warnings are left: https://github.com/freifunk/meshviewer/issues/49

> Also I'm wondering if there is an easy option to keep the whole graph centred all the time, even before it gets rescaled to zoom to a specific node?

Can you fill in details here https://github.com/freifunk/meshviewer/issues/50 ?

MyIgel commented on README.md:
````diff
@@ -1,49 +1,64 @@
+  node npm run dev -- --host 0.0.0.0
 ```
-The map is reachable at [localhost:3000](http://localhost:3000).
 
````
```suggestion
The map is reachable at [localhost:5173](http://localhost:5173).

```

MyIgel reviewed (changes requested):
That looks better, just a small hickup:

MyIgel commented:
After switchong from the node graph http://localhost:8080/#!/en/graph/9cc7a67853ea back to the map view i got a

```
Uncaught TypeError: u is null
    Y index-592e0aaf.js:17419
    y index-592e0aaf.js:17421
    setTimeout handler*Y index-592e0aaf.js:17420
    y index-592e0aaf.js:17421
    setTimeout handler*Y index-592e0aaf.js:17420
    y index-592e0aaf.js:17421
    setTimeout handler*Y index-592e0aaf.js:17420
    y index-592e0aaf.js:17421
    setTimeout handler*Y index-592e0aaf.js:17420
    y index-592e0aaf.js:17421
    setTimeout handler*Y index-592e0aaf.js:17420
    gotoNode index-592e0aaf.js:17594
    f index-592e0aaf.js:12766
    f index-592e0aaf.js:12765
    w index-592e0aaf.js:12800
    resolve index-592e0aaf.js:12539
    Dn index-592e0aaf.js:12433
    resolve index-592e0aaf.js:12538
    Dn index-592e0aaf.js:12433
    resolve index-592e0aaf.js:12535
    _onLocationChange index-592e0aaf.js:12732
    _listen index-592e0aaf.js:12711
    qt index-592e0aaf.js:12313
    gg index-592e0aaf.js:12814
    p1 index-592e0aaf.js:22913
    m1 index-592e0aaf.js:22961
    _1 index-592e0aaf.js:22963
    sd index-592e0aaf.js:1
    <anonymous> index-592e0aaf.js:22966
```

in `if (clearTimeout(y), !st && u.alpha() > 0.3) {` of
```
I = 3,
      T = 0.01;
      Ht.setTransform(f);
      function z() {
        a.width = o.offsetWidth,
        a.height = o.offsetHeight,
        Ht.setMaxArea(a.width, a.height)
      }
      function F(q) {
        f.x = q.x,
        f.y = q.y,
        f.k = q.k
      }
      function Y(q, st) {
        if (clearTimeout(y), !st && u.alpha() > 0.3) {
          y = setTimeout(function () {
            Y(q)
          }, 300);
          return
        }
        var A = q(),
        $ = A[0],
        E = A[1],
        O = A[2],
        D = {
          k: O
        };
        D.x = (a.width + e.getWidth()) / 2 - $ * O,
        D.y = a.height / 2 - E * O;
        var W = {
          x: f.x,
          y: f.y,
          k: f.k
        },
        R = Rh(W, D),
        G = po(
          function (K) {
            if (K >= b) {
              G.stop();
              return
            }
            var et = R(zy(K / b));
            F(et),
            window.requestAnimationFrame(U)
          }
        )
      }
      function X(q) {
```

MyIgel commented on public/config.json:
```diff
@@ -0,0 +1,366 @@
+{
```
The fact that this file must exist and gets overwritten in the `build` dir complicates the development workflow as it means that developers have to overwrite it when developing with their own community maps or copy it after every single build if i see that correctly?

Imho the easiest  option here would be to exclude it from the deletion when wiping the build dir?

weeman1337 replied:
This file should not be tracked by Git. I will remove it and update the docs. Like before you have to get a config file from somewhere.

weeman1337 replied:
https://github.com/freifunk/meshviewer/compare/b113ad7e60984d59be888c2a520aff6749e067e5..8e06242e20cb7679ed27e5764434e557c2181244

MyIgel commented on vite.config.js:
```diff
@@ -0,0 +1,55 @@
+    },
+  },
+  build: {
+    outDir: "build",
```
```suggestion
    outDir: "build",
    emptyOutDir: false,
```
might be an option eventually (and removing the config file from public)?

weeman1337 replied:
It should be removed from public but I would not recommend to keep state in the output directory.

MyIgel reviewed (approved):
@weeman1337 do you know what causes the warning `downloadable font: Glyph bbox was incorrect (glyph ids 7 8 10 11 13) (font-family: "ionicons" style:normal weight:400 stretch:100 src index:0) source: data:font/woff2;base64,d09GMgABAAAAAAkgAAsAAAAAEOAAAAjRAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAABmAAgXQRCAqXXJIPCzgAATYCJANsBCAFgyoHIBsaD … fi2GtC5hmIr7PrEs0CQmA6tg57/RqluXMBqJkrfTjXXoL6VGgqluVrNBHWp7mZ9z230FJsur7/6fLKMh+/hgYALNCWHXUUOtjDg0lIQ6S1hCnvowFFEmM/dEImsAAAA= ` on firefox  and if its somehow fixable?

skorpy2009 reviewed (approved):
LGTM
works for me

skorpy2009 commented:
Danke @weeman1337
